agentMET4FOF/metrological_streams.py

```diff
@@ -144,8 +152,12 @@ def _default_uncertainty_generator(
             constant time and value uncertainties each of the same shape
             as ``time``
         """
+        if not self.exp_unc:
```

Comment:
I am wondering if this condition is correct. If we actually want to check if `self.exp_unc` is not None, then we should do this more explicitly. The current condition is True for `self.exp_unc == 0` and `False` for any other float for instance. 
```suggestion
        if self.exp_unc is not None:
```

Reply:
Additionally I think, we should actually do this check and properly set `value_unc` in the constructor.

Reply:
Good suggestion. I've changed it in the next commit.